tests/test_vqe.py

```diff
@@ -639,6 +689,17 @@ def test_hamiltonian_str(self, terms, string):
         H = qml.vqe.Hamiltonian(*terms)
         assert H.__str__() == string
 
+    @pytest.mark.parametrize("terms, string", zip(valid_hamiltonians, valid_hamiltonians_repr))
+    def test_hamiltonian_repr(self, terms, string):
+        """Tests that the __repr__ function for printing is correct"""
+        H = qml.vqe.Hamiltonian(*terms)
+        assert H.__repr__() == string
```

Comment:
Glad that we have a test for this, but don't worry too much about the codecov error in the `__repr__` method above -- this is a known bug